lib/session.js

```diff
@@ -115,10 +133,25 @@ exports.signin = function(lookup, req, res, onSuccess, onFail) {
 
 exports.signout = function(req, res, next) {
 
-	res.clearCookie('keystone.uid');
-	req.user = null;
+	keystone.callHook(req.user, 'pre:signout', function(err) {
+		if (err) {
+			console.log("An error occurred in signout 'pre' middleware", err);
```

Comment:
I think that solution is good,  in terms of a graceful handling of a terminating error, its a good outcome; and this definitely makes it more [obviously] ugly, which typically means it will likely to be addressed better later haha.

Any reason for the 2nd non-terminating error in the post hook?